### PR TITLE
Add api

### PR DESCRIPTION
**Pull request by safar1212**
In this pull request:

I added the addition and deletion functionality through API

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,33 +1,76 @@
...
+//       title, author, category, item_id,
+//     } = book[0];
+//     return {
+//       item_id, title, author, category,
+//     };
+//   });
+//   return objectOfBooks;
+// });
+
+// reducer
```
Comment (GzimAsani) [inline src/redux/books/books.js]: - [ ] It is not a good idea to leave the commented-out code. It makes your code looks unprofessional

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,33 +1,76 @@
...
+          item_id: key,
+          author: book[key][0].author,
+          title: book[key][0].title,
+          category: book[key][0].category,
+        });
+      });
+
+      dispatch({ type: GET_BOOKS, booksArray });
+    });
+};
```
Comment (GzimAsani) [inline src/redux/books/books.js:49]: - [ ] It is a project requirement to use `createAsyncThunk` in the project for creating thunk action creators. Kindly do that

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @safar1212      :wave:   
 Good job so far, ! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good work, your project works as expected  :+1: 
 
 ## Required Changes 

- [x] _Check the comments below :arrow_double_down:_



 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 2 above)
Comment (safar1212) [reply src/redux/books/books.js:49]: @GzimAsani Thanks for your comments. Actually, I was trying to implement `createAsyncThunk`, but it wasn't working. Therefore, I commented that out. Moreover, it isn't a requirement. Please check out the screenshot below. 
It says **Fetch OR Axios with createAsyncThunk**. We can use fetch as well.

Waiting for your reply.

Comment (GzimAsani) [reply src/redux/books/books.js:49]: Hello @safar1212, I apologize I read it wrong I missed the `OR` part, kindly remove the commented out code and you should be good :+1:

**Review by keneogu — APPROVED** [review]
Hi @safar1212,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**